File: 프로그래머스/1/42840. 모의고사/모의고사.js
//1. 최대치로 생각했을 때 문제가 10000문제 

// 2. while 문으로 
 // 가장 많은 문제를 맞힌 사람이 누구인지 배열에 담아 return 하도록 solution 함수를 작성해주세요.
function solution(answers) {
    let answer = [];
    let  F_1 = [1, 2, 3, 4, 5, 1, 2, 3, 4, 5] ,F_2 = [2, 1, 2, 3, 2, 4, 2, 5, 2, 1, 2, 3, 2, 4, 2, 5] , F_3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5, 3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    let i = 0
    let cntTotal = { 
        1: 0 ,
        2 : 0,
        3 :0 
    }
    while(i<answers.length){
        
        if(F_1[i% F_1.length]===answers[i]){
            cntTotal["1"]++
        }
        if(F_2[i% F_2.length]===answers[i]){
            cntTotal["2"]++
        }
        if(F_3[i% F_3.length]===answers[i]){
            cntTotal["3"]++
        }
        
        i++
        
    }
       
    const MAX = Math.max(...Object.values(cntTotal) )
    
    for(const f in cntTotal){
        if(cntTotal[f]===MAX){
            answer.push(Number(f))
        }
        
    }
   
 
    return answer
}

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.* ; 
class Solution {
    public List<Integer> solution(int[] answers) {
        int[] answer = {};
        int [] f_1 = {1, 2, 3, 4, 5, 1, 2, 3, 4, 5} ; 
        int [] f_2 = {2, 1, 2, 3, 2, 4, 2, 5, 2, 1, 2, 3, 2, 4, 2, 5} ; 
        int [] f_3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5, 3, 3, 1, 1, 2, 2, 4, 4, 5, 5} ; 
        
         
        
       int [] people3 ={0,0,0};
        
        for(int i = 0 ; i<answers.length; i++){
            
            if(f_1[i%f_1.length]==answers[i]){
                people3[0]+=1 ;
            }
             if(f_2[i%f_2.length]==answers[i]){
                people3[1]+=1 ;
            }
             if(f_3[i%f_3.length]==answers[i]){
                people3[2]+=1 ;
            }
        }
        
        // for(int k : people3){
        //     System.out.println(k) ; 
        // }
        
        
        int maxScore = Math.max(people3[0],  Math.max(people3[1], people3[2])) ; 
        
          List<Integer> topScorers = new ArrayList<>();
        
        // 최대 점수를 받은 사람들 찾기
        for (int i = 0; i < people3.length; i++) {
            if (people3[i] == maxScore) {
                topScorers.add(i + 1);  // 인덱스를 1부터 시작하도록 조정
            }
        }
        
        // System.out.println(topScorers);
        
        
        
        
        
        
        return topScorers;
    }
}